src/loader.js

```diff
@@ -203,11 +204,11 @@ async function loader(content) {
     }
   }
 
-  let isAbsolute = isAbsoluteURL(this.resourcePath);
-
-  const filename = isAbsolute
-    ? this.resourcePath
-    : path.relative(this.rootContext, this.resourcePath);
+  const filename =
+    ABSOLUTE_URL_REGEX.test(this.resourcePath) &&
+    !WINDOWS_PATH_REGEX.test(this.resourcePath)
+      ? this.resourcePath
+      : this.utils.contextify(this.rootContext, this.resourcePath);
```

Comment:
@alexander-akait

The `filename` must be a `relative path` without leading `./`:

- this.utils.contextify(this.rootContext, this.resourcePath) => `./src/image.svg` (❌ wrong filename)
- path.relative(this.rootContext, this.resourcePath) => `src/image.svg` (✅ correct filename)

Please, see the details in my [updated comment](https://github.com/webpack-contrib/image-minimizer-webpack-plugin/issues/458#issuecomment-2545320268)
I have linked the NEW small as possible [repo](https://github.com/webdiscus/issue-cache-filesystem-rebuild-image-minimizer) with reproducible issue.

Thank you!

Reply:
this PR doesn't fix the issue

Reply:
strange, I'll have a look now

Reply:
https://github.com/webpack-contrib/image-minimizer-webpack-plugin/releases/tag/v4.1.3

Reply:
Now with tests :smile:

Reply:
@alexander-akait 

it works now :-)
thank you!